sdk/aqueduct/models/integration.py

```diff
@@ -76,6 +77,34 @@ def name(self) -> str:
     def type(self) -> ServiceType:
         return self._metadata.service
 
+    def _validate_is_connected(self) -> None:
+        """Method used to determine if this integration was successfully connected to or not.
+        If not successfully connected (or pending), we will raise an Exception.
+        """
+        # TODO(ENG-2813): Remove the assumption that a missing `exec_state` means success.
+        if (
+            self._metadata.exec_state is None
+            or self._metadata.exec_state.status == ExecutionStatus.SUCCEEDED
+        ):
+            return
+
+        if self._metadata.exec_state.status == ExecutionStatus.FAILED:
+            assert self._metadata.exec_state.error is not None
+            raise IntegrationFailedToConnect(
+                "The attempt to connect to %s has failed with error: %s\n%s\n\n "
+                "Please see the /integrations page on the UI for more details."
+                % (
+                    self.name(),
+                    self._metadata.exec_state.error.tip,
+                    self._metadata.exec_state.error.context,
+                )
+            )
+        else:
```

Comment:
Is there an explicit `ExecutionStatus.Pending` status to check for in progress, or is it just that after 2813, all statuses that are `None` are considered in progress?

Reply:
Nah I don't think this has to do with ENG-2813 - that's just saying that the None case will no longer exist. I don't think we ever have a pending state, we just go directly to running. Can make that a harder assertion here.

Reply:
Actually, I decided to simply document this assumption. The assertion seems too aggressive since this isn't a code path we are really testing right now.

Reply:
sounds good